Document Destroy Async (#41680)
rails/rails: guides/source/association_basics.md

associated objects.
* `:delete`, when the object is destroyed, all its associated objects will be
deleted directly from the database without calling their `destroy` method.

> * `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob`

Reviewer: ```suggestion
* `:destroy_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob`
```

> job is enqueued to which will call destroy on its associated objects. ActiveJob must be setup

Reviewer 2: ```suggestion
job is enqueued to which will call destroy on its associated objects. Active Job must be setup
```

> job is enqueued to which will call destroy on its associated objects. Active Job must be setup

Reviewer: ```suggestion
job is enqueued to which will call destroy on its associated objects. Active Job must be set up
```

* `:destroy` causes the associated object to also be destroyed
* `:delete` causes the associated object to be deleted directly from the database (so callbacks will not execute)

> * `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. ActiveJob must be setup for this to work.

Reviewer 2: ```suggestion
* `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be setup for this to work.
```

> * `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be setup for this to work.

Reviewer: ```suggestion
* `:destroy_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be set up for this to work.
```

* `:destroy` causes all the associated objects to also be destroyed
* `:delete_all` causes all the associated objects to be deleted directly from the database (so callbacks will not execute)

> * `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. ActiveJob must be setup for this to work.

Reviewer 2: In our docs, when we talk about a library we tend to use the library name rather than the module name (ie Active Job over ActiveJob).

```suggestion
* `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be setup for this to work.
```

> * `:destory_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be setup for this to work.

Reviewer: ```suggestion
* `:destroy_async`: when the object is destroyed, an `ActiveRecord::DestroyAssociationAsyncJob` job is enqueued to which will call destroy on its associated objects. Active Job must be set up for this to work.
```
